### PR TITLE
Add multiple instantiation protection

### PR DESCRIPTION
**Pull request by udaydenduluri33**
https://tools.hmcts.net/jira/browse/EUI-3774

The feature toggle service singleton implementation.

### CODE REVIEW

**Review by leepavlouhmcts — APPROVED** [review]
looks good

File: projects/exui-common-lib/src/lib/services/helpers/rootInjectorGuard.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+export class RootInjectorGuard {
+    constructor(type: Type<any>) {
+        const parent = inject(type, InjectFlags.Optional | InjectFlags.SkipSelf);
+
+        if (parent) {
+            throw Error(`[${type}]: Creating multiple instances, but should be singleton.`);
+        }
+    }
+}
```
Comment (connorpgpmcelroy) [inline projects/exui-common-lib/src/lib/services/helpers/rootInjectorGuard.ts:11]: Should there be newline here?

File: projects/exui-common-lib/src/lib/window.ts [hunk 2]
```diff
@@ -1,4 +1,3 @@
 import { InjectionToken } from '@angular/core';
 
-export const windowToken = new InjectionToken('Window');
-export function windowProvider() { return window; }
+export const windowToken = new InjectionToken('Window', { providedIn: 'root', factory: () => window });
```
Comment (connorpgpmcelroy) [inline projects/exui-common-lib/src/lib/window.ts:3]: Also, newline here?

**Review by connorpgpmcelroy — COMMENTED** [review]
Just needs some new lines at end of two files and then I can approve

File: projects/exui-common-lib/src/lib/services/helpers/rootInjectorGuard.ts (hunk 1 above)
Comment (udaydenduluri33) [reply projects/exui-common-lib/src/lib/services/helpers/rootInjectorGuard.ts:11]: added

File: projects/exui-common-lib/src/lib/window.ts (hunk 2 above)
Comment (udaydenduluri33) [reply projects/exui-common-lib/src/lib/window.ts:3]: New line is already present

**Review by connorpgpmcelroy — APPROVED** [review]
Approved on the basis of the new ticket created, EUI-3802

**Pull request closed without merging**